src/java/test/bin/test-github-pr.sh

```diff
@@ -0,0 +1,612 @@
+      if [[ $# != 10 ]] ; then
+        echo "ERROR: usage $0 DEVELOPER <GIT_PR_URL> <SCRATCH_DIR> <GIT_CMD> <GREP_CMD> <PATCH_CMD> <FINDBUGS_HOME> <FORREST_HOME> <WORKSPACE_BASEDIR> <JAVA5_HOME>"
+        cleanupAndExit 0
+      fi
+      PATCH_DIR=$3
+      PATCH_FILE=${PATCH_DIR}/patch
+      curl -L $2.diff > ${PATCH_FILE}
+      ### PATCH_FILE contains the location of the patchfile
+      if [[ ! -e "$PATCH_FILE" ]] ; then
+        echo "Unable to locate the patch file $PATCH_FILE"
+        cleanupAndExit 0
+      fi
+      ### Check if $PATCH_DIR exists. If it does not exist, create a new directory
+      if [[ ! -e "$PATCH_DIR" ]] ; then
+	mkdir "$PATCH_DIR"
+	if [[ $? == 0 ]] ; then
+	  echo "$PATCH_DIR has been created"
+	else
+	  echo "Unable to create $PATCH_DIR"
+	  cleanupAndExit 0
+	fi
+      fi
+      GIT=$4
+      GREP=$5
+      PATCH=$6
+      FINDBUGS_HOME=$7
+      FORREST_HOME=$8
+      BASEDIR=$9
+      JAVA5_HOME=${10}
+      ### Obtain the patch filename to append it to the version number
+      local subject=`grep "Subject:" ${PATCH_FILE}`
+      local length=`expr match ${subject} ZOOKEEPER-[0-9]*`
+      local position=`expr index ${subject} ZOOKEEPER-`
+      defect=${${subject:$position:$length}#ZOOKEEPER-}
+      ;;
+    *)
+      echo "ERROR: usage $0 QABUILD [args] | DEVELOPER [args]"
+      cleanupAndExit 0
+      ;;
+  esac
+}
+
+###############################################################################
+checkout () {
+  echo ""
+  echo ""
+  echo "======================================================================"
+  echo "======================================================================"
+  echo "    Testing patch for pull request ${PULLREQUEST_ID}."
+  echo "======================================================================"
+  echo "======================================================================"
+  echo ""
+  echo ""
+  ### When run by a developer, if the workspace contains modifications, do not continue
+  # Ref http://stackoverflow.com/a/2659808 for details on checking dirty status
+  ${GIT} diff-index --quiet HEAD
+  if [[ $? -ne 0 ]] ; then
+    uncommitted=`${GIT} diff --name-only HEAD`
+    uncommitted="You have the following files with uncommitted changes:${NEWLINE}${uncommitted}"
+  fi
+  untracked="$(${GIT} ls-files --exclude-standard --others)" && test -z "${untracked}"
+  if [[ $? -ne 0 ]] ; then
+    untracked="You have untracked and unignored files:${NEWLINE}${untracked}"
+  fi
+
+  if [[ $QABUILD == "false" ]] ; then
+    if [[ $uncommitted || $untracked ]] ; then
+      echo "ERROR: can't run in a workspace that contains the following modifications"
+      echo ""
+      echo "${uncommitted}"
+      echo ""
+      echo "${untracked}"
+      cleanupAndExit 1
+    fi
+  else
+    # I don't believe we need to do anything here - the jenkins job will
+    # cleanup the environment for us ("cleanup before checkout" action)
+    # on the precommit jenkins job
+    echo
```

Comment:
is this echo here just to get around the syntax error? if so you can use : instead